flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputSplit.java

```diff
@@ -19,28 +19,19 @@
 
 package org.apache.iceberg.flink.source;
 
-import org.apache.flink.core.io.InputSplit;
 import org.apache.flink.core.io.LocatableInputSplit;
 import org.apache.iceberg.CombinedScanTask;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
 
-/**
- * TODO Implement {@link LocatableInputSplit}.
- */
-public class FlinkInputSplit implements InputSplit {
+public class FlinkInputSplit extends LocatableInputSplit {
 
-  private final int splitNumber;
   private final CombinedScanTask task;
 
-  FlinkInputSplit(int splitNumber, CombinedScanTask task) {
-    this.splitNumber = splitNumber;
+  FlinkInputSplit(int splitNumber, CombinedScanTask task, String[] hosts) {
```

Comment:
What value will `hosts` be if locality is disabled? Simply `null`?

Reply:
yes, It's a `null`. And `org.apache.flink.core.io.LocatableInputSplit#LocatableInputSplit(int, java.lang.String[])` converts `null `to a empty array.

refer
https://github.com/apache/flink/blob/137b65c5f23e5ba546ce52bd15fa7aa528ab95de/flink-core/src/main/java/org/apache/flink/core/io/LocatableInputSplit.java#L61-L64

Maybe add an @Nullable to it? like this:
```
  FlinkInputSplit(int splitNumber, CombinedScanTask task, @Nullable String[] hosts) {
    super(splitNumber, hosts);
    this.task = task;
  }

```